### PR TITLE
templ: init at 0.2.476

### PR DESCRIPTION
**Pull request by luleyleo**
## Description of changes

[templ](https://github.com/a-h/templ): A language for writing HTML user interfaces in Go. 

Note: I didn't understand how choosing a maintainer works, so I left the maintainers field empty at first.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/tools/templ/default.nix
Comment (nevivurn) [inline pkgs/development/tools/templ/default.nix:44]: What are these tags for?

File: pkgs/development/tools/templ/default.nix
Comment (nevivurn) [inline pkgs/development/tools/templ/default.nix:16]: trimpath is added by buildGoModule [by default](https://github.com/NixOS/nixpkgs/blob/a1422a7f8f12b68e307b92b7daa254f412e9ff96/pkgs/build-support/go/module.nix#L154), this can be removed.

File: pkgs/development/tools/templ/default.nix
Comment (nevivurn) [inline pkgs/development/tools/templ/default.nix:46]: The tests seem to pass just fine, this should be removed.

File: pkgs/development/tools/templ/default.nix
Comment (nevivurn) [inline pkgs/development/tools/templ/default.nix:52]: If you're willing to maintain this package, you can add yourself to the maintainers list by

1. Adding yourself to `maintainers/maintainer-list.nix` on a separate commit like `maintainers: add luleyleo`
2. Add yourself to the list here.

Otherwise, feel free to add me (`nevivurn`) as a maintainer.

**Review by nevivurn — COMMENTED** [review]
Thanks for the contribution, just a few comments.

File: pkgs/development/tools/templ/default.nix
Comment (luleyleo) [reply pkgs/development/tools/templ/default.nix]: Ah, they were leftovers from the app I based this off from. I've removed them.

File: pkgs/development/tools/templ/default.nix
Comment (luleyleo) [reply pkgs/development/tools/templ/default.nix]: I've added myself to the list now.

**Review by Janik-Haag — COMMENTED** [review]
Please squash all the `templ:*` commits into `templ: init at 0.2.476` since nixpkgs tries to keep a atomic commit history. This looks good to me other then that.

**Review by nevivurn — COMMENTED** [review]
Also LGTM, just needs squashing as mentioned.

**Review by Janik-Haag — APPROVED** [review]

**Comment by Janik-Haag** [comment]
just waiting for ci now.

**Pull request merged**
